/*
  CarStandFCOOL API

  An API for managing a cloud-based car dealership service

  The version of the OpenAPI document: 1.0.0

  Generated by OpenAPI Generator: https://openapi-generator.tech
*/

syntax = "proto3";

package openapitools.services.maintenanceservice;

import "google/protobuf/empty.proto";
import public "models/maintenance.proto";

service MaintenanceService {
  rpc MaintenanceCreate (MaintenanceCreateRequest) returns (Maintenance);

  rpc MaintenanceDelete (MaintenanceDeleteRequest) returns (google.protobuf.Empty);

  rpc MaintenanceReadAll (google.protobuf.Empty) returns (MaintenanceReadAllResponse);

  rpc MaintenanceReadOne (MaintenanceReadOneRequest) returns (Maintenance);

  rpc MaintenanceUpdate (MaintenanceUpdateRequest) returns (Maintenance);

}

message MaintenanceCreateRequest {
  // Maintenance to create
  Maintenance maintenance = 1;

}

message MaintenanceDeleteRequest {
  // ID of the maintenance
  int32 maintenanceId = 1;

}

message MaintenanceReadAllResponse {
  repeated Maintenance data = 1;
}

message MaintenanceReadOneRequest {
  // ID of the maintenance
  int32 maintenanceId = 1;

}

message MaintenanceUpdateRequest {
  // ID of the maintenance
  int32 maintenanceId = 1;
  Maintenance maintenance = 2;

}

